pandas/core/internals/concat.py

```diff
@@ -61,8 +62,19 @@ def concatenate_block_managers(
                 values = values.view()
             b = b.make_block_same_class(values, placement=placement)
         elif _is_uniform_join_units(join_units):
-            b = join_units[0].block.concat_same_type([ju.block for ju in join_units])
-            b.mgr_locs = placement
+            blk = join_units[0].block
+            vals = [ju.block.values for ju in join_units]
+
+            if not blk.is_extension:
+                values = concat_compat(vals, axis=blk.ndim - 1)
```

Comment:
hmm, doesn't concat_compat already handle this? why do you need cases here?

Reply:
> why do you need cases here?

I hope you're sitting down because this may come as a shock: because EAs dont support 2D.

Reply:
i get that
but the cases are all the same

Reply:
The last one is not, we dont pass `axis`

Reply:
ok, but you still don't need line 70 at all

Reply:
sure, will update

Reply:
updated+green